docs: add MySQL 8.4 migration guide for customers (#10163)
pantheon-systems/documentation: src/source/content/guides/mariadb-mysql/11-mysql-84.md

---
title: MariaDB and MySQL on Pantheon
subtitle: MySQL 8.4
description: Learn how to migrate from MariaDB to MySQL 8.4 on Pantheon.
contenttype: [guide]
innav: [false]
categories: [database]
cms: [--]
audience: [development]
product: [--]
integration: [--]
tags: [database]
showtoc: true
permalink: docs/guides/mariadb-mysql/mysql-84
reviewed: "2026-07-13"
---

Pantheon supports MySQL 8.4 LTS as a database engine alongside MariaDB. MySQL 8.4 offers long-term support from Oracle through 2032.

## Before You Begin
### Considerations
Before enabling MySQL 8.4, consider the following limitations:

> - **No self-service rollback.** Reverting from MySQL 8.4 to MariaDB requires Pantheon support. Always test on a [Multidev](/guides/multidev) first.

Reviewer: ```suggestion
- **No self-service rollback.** Reverting from MySQL 8.4 to MariaDB requires Pantheon support. Always test on a [Multidev](/guides/multidev) or Dev environment before switching your Live environment.
```

> - **Backups are engine-specific.** A backup taken on MySQL 8.4 cannot be restored to a MariaDB environment, and vice versa.

Reviewer: Why can't a backup taken in MariaDB be imported to a MySQL environment? aren't we doing this conversion with a backup and import?

Author: You are right that the migration itself uses backup and import internally, and the platform handles the engine conversion during that process. The limitation is that a manual backup/restore initiated by the customer does not do that conversion. Rewrote the bullet to explain both sides.

> - **MariaDB 10.6 required.** Your site must be on MariaDB 10.6 before migrating to MySQL 8.4. Sites on older MariaDB versions will be upgraded to 10.6 first automatically.

Reviewer: @kanishkaverma in Slack we discussed making this a recommendation rather than a requirement. Let's hold off on publishing until we know for certain.

Author: Softened to "recommended" with a note that the migration currently requires 10.6 as the source version. That way the doc is accurate to today's behavior but doesn't read as a hard prerequisite for customers planning ahead.

### CMS Version Compatibility
Before enabling MySQL 8.4, verify the following:

| CMS | Minimum Version | Notes |
|-----|----------------|-------|
| WordPress 6.x | All versions | Fully compatible |
| WordPress 5.x | 5.9+ | Older versions may hit reserved word or sql_mode issues |

> | Drupal 10 | 10.2+ | Earlier versions may need patches for the `GROUPS` reserved word |

Reviewer: What patches would they need? It might be wiser for us to just recommend upgrading to the highest available point release.

Author: Good call. Changed both WordPress 5.x and Drupal 10 to recommend upgrading to the latest point release rather than pinning specific versions.

| Drupal 9 | 9.5+ | Community support ended |
| Drupal 7 | 7.x | Works with caveats. See [ONLY_FULL_GROUP_BY](#only_full_group_by-errors) below |

### Reserved Word Conflicts

MySQL 8.4 added several reserved words. If your database uses any of these as table or column names, queries will fail unless the names are quoted with backticks:

- `GROUPS` (common in Drupal sites using the Groups module)
- `RANK`, `DENSE_RANK`, `ROW_NUMBER`
- `JSON_TABLE`, `LATERAL`, `RECURSIVE`

```sql
-- This breaks on MySQL 8.4 if "groups" is a table name
SELECT * FROM groups;

-- This works on both MariaDB and MySQL 8.4
SELECT * FROM `groups`;
```

## What To Expect
### Migration Timing

| Database Size | Estimated Total Time |
|--------------|---------------------|
| Under 500 MB | 3-5 minutes |
| 500 MB - 2 GB | 5-10 minutes |
| 2 - 10 GB | 10-20 minutes |
| 10 - 30 GB | 20-60 minutes |
| 30 - 100 GB | 1-3 hours |

> ### During Migration

Reviewer: ```suggestion
### During migration
```

1. A new MySQL 8.4 database is provisioned for your environment.
2. Your existing MariaDB data is exported and imported into the new MySQL 8.4 database.
3. The platform verifies the data transferred correctly.
4. The new MySQL 8.4 database is promoted as your active database.

Your site's database is **read-only during the export** and **briefly unavailable during the switchover**.

> ### What Changes after migration

Reviewer: ```suggestion
### What changes after migration
```

- `SELECT VERSION()` returns `8.4.x` instead of a MariaDB version string.
- The default collation for new tables is `utf8mb4_0900_ai_ci`. Migrated tables keep their original collation (`utf8mb4_general_ci`).
- Stricter SQL mode enforcement is enabled by default.

> ### What Stays the Same after migration

Reviewer: ```suggestion
### What stays the same after migration
```

- Connection credentials (host, port, username, password).
- Database name (`pantheon`).
- All your data, tables, and indexes.
- Backup and restore workflows (within the same engine).

> ## How to Enable MySQL 8.4

Reviewer: ```suggestion
## How to enable MySQL 8.4
```

> Add the following to your site's `pantheon.yml` file:

Reviewer: Before we merge this PR we need to figure out which upstreams have are getting their pantheon.upstream.yml changed on which dates

Author: Added an info alert below the enablement section explaining that managed upstreams will get the pantheon.upstream.yml change on a per-upstream schedule from Pantheon, and that customers only need to act if they are on a custom upstream or want to opt in early on a non-production environment. The specific upstream dates can be filled in closer to launch -- this framing works regardless of the exact schedule.

```yaml:title=pantheon.yml
database:

>   version: 8.4

Reviewer: ```suggestion
  type: mysql
  version: 8.4
```
